src/tools/build-manifest/src/main.rs

```diff
@@ -402,11 +422,7 @@ impl Builder {
                         Some(p) => p,
                         None => return false,
                     };
-                    let target = match pkg.target.get(&c.target) {
-                        Some(t) => t,
-                        None => return false,
-                    };
-                    target.available
```

Comment:
This line was added in https://github.com/rust-lang/rust/pull/49705 to help mitigate https://github.com/rust-lang/rust/issues/49462. Just wanted to confirm, but you think that the bug won't be coming back with this change? I haven't checked recently how rigorously rustup checks all these flags...

Reply:
Ok, no, with this change that bug would probably come up again. We need an explicit list of targets with docs and pass it [here](https://github.com/rust-lang/rust/blob/f1a5373a1b6222cf510d410f37278012b096fbfb/src/tools/build-manifest/src/main.rs#L321) instead of all the targets.

Reply:
Pushed a commit to explicitly package docs only for tier 1 platforms.